pkg/types/types.go

```diff
@@ -70,12 +116,40 @@ type RetrievalResult struct {
 	Err   error
 }
 
-type CandidateRetrievalCall struct {
-	Candidates         []RetrievalCandidate
-	CandidateRetrieval CandidateRetrieval
+var _ RetrievalTask = AsyncRetrievalTask{}
+
+// AsyncRetrievalTask runs an asynchronous retrieval and returns a result
+type AsyncRetrievalTask struct {
+	Candidates              InboundAsyncCandidates
+	AsyncCandidateRetrieval CandidateRetreival
 }
 
-type RetrievalCoordinator func(context.Context, []CandidateRetrievalCall) (*RetrievalStats, error)
+// Run executes the asychronous retrieval task
+func (acrc AsyncRetrievalTask) Run() (*RetrievalStats, error) {
+	return acrc.AsyncCandidateRetrieval.RetrieveFromAsyncCandidates(acrc.Candidates)
```

Comment:
```suggestion
func (art AsyncRetrievalTask) Run() (*RetrievalStats, error) {
	return art.AsyncCandidateRetrieval.RetrieveFromAsyncCandidates(art.Candidates)
```

doesn't really matter, but you can see remnants of the original naming here that might be confusing